bundles/binding/org.openhab.binding.panstamp/src/main/java/org/openhab/binding/panstamp/internal/PanStampConversions.java

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.panstamp.internal;
+
+import me.legrange.panstamp.Endpoint;
+
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+
+/**
+ * Some utility methods to convert date between different realms and types.
+ * 
+ * @author Gideon le Grange
+ * @since 1.8.0
+ */
+class PanStampConversions {
+
+	static String toString(Endpoint<?> ep) {
+		return String.format("%d->%d/%s", ep.getRegister().getDevice()
+				.getAddress(), ep.getRegister().getId(), ep.getName());
+	}
+
+	/** 
+	 * convert the value value for and endpoint to a state message 
+	 * @param ep The endpoint for which the value is converted 
+	 * @param val The value being converted 
+	 * @return a new State object for the value
+	 */
+	static State toState(Endpoint<?> ep, Object val) throws ValueException {
+		switch (ep.getType()) {
+			case STRING:
+				return new StringType(val.toString());
+			case NUMBER:
+				return new DecimalType(((Double) val));
+			case INTEGER:
+				return new DecimalType(((Long) val));
+			case BINARY:
+				return ((Boolean) val) ? OnOffType.ON : OnOffType.OFF;
+			default:
+				throw new ValueException(
+					String.format(
+							"Unsupported panStamp endpoint type '%s' for endpoint %s. BUG!",
+							ep.getType(), toString(ep)));
+		}
+	}
+
+	/** 
+	 * Convert a command to a String value
+	 * @param cmd The command to convert to a String 
+	 * @return The string 
+	 * @throws ValueException If a conversion isn't possible
+	 */
+	static String toString(Command cmd) throws ValueException {
+		if (cmd instanceof StringType) {
+			return ((StringType) cmd).toString();
+		}
+		throw new ValueException(String.format(
+				"Unsopported command type '%s' for string endpoint", cmd
+						.getClass().getSimpleName()));
+	}
```

Comment:
Since a `StringType` command could contain any string data including junk, then I am missing how `DecimalType.toString()` could be worse than `StringType.toString()`.  But if my suggestion could introduce junk data to the network, then definitely leave it as is.